app/decorators/controllers/hyrax/downloads_controller_decorator.rb

```diff
@@ -0,0 +1,30 @@
+module Hyrax
+  DownloadsController.class_eval do
+    # `asset` is inherited from hydra-head/hydra-core/app/controllers/concerns/hydra/controller/download_behavior.rb
+    def send_content
+      super
+      update_download_stats(asset)
+    end
+
+    def download_stats(current_file)
+      if FileDownloadStat.find_by(file_id: current_file.id)
+        FileDownloadStat.find_by(file_id: current_file.id)
```

Comment:
One common practice to avoid long `if/else` blocks is do to:
```ruby
stat = FileDownloadStat.find_by(file_id: current_file.id)
return stat if stat
... (the stuff that was in the else block)...
```

Reply:
:100: